pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1334,13 +1334,27 @@ func (c *PodAffinityChecker) satisfiesExistingPodsAntiAffinity(pod *v1.Pod, meta
 //  topology of all the "terms" for the given "pod".
 func (c *PodAffinityChecker) nodeMatchesAllTopologyTerms(pod *v1.Pod, topologyPairs *topologyPairsMaps, nodeInfo *schedulercache.NodeInfo, terms []v1.PodAffinityTerm) bool {
 	node := nodeInfo.Node()
-	for _, term := range terms {
-		if topologyValue, ok := node.Labels[term.TopologyKey]; ok {
-			pair := topologyPair{key: term.TopologyKey, value: topologyValue}
-			if _, ok := topologyPairs.topologyPairToPods[pair]; !ok {
-				return false
+	for idx, term := range terms {
+		topologyValue, ok := node.Labels[term.TopologyKey]
+		if !ok {
+			return false
+		}
+
+		pair := topologyPair{key: term.TopologyKey, value: topologyValue}
+		podSet, ok := topologyPairs.topologyPairToPods[pair]
+		if !ok {
+			return false
+		}
+		// "idx" should be found in "podSet[<any pod>]"
+		found := false
```

Comment:
the `found` variable/logic could be omitted by using a continue <outer loop> trick.